libobs/obs-cocoa.m

```diff
@@ -889,8 +894,48 @@ bool obs_hotkeys_platform_is_pressed(obs_hotkeys_platform_t *plat,
 	return plat->is_key_down[key];
 }
 
+static void mach_set_thread_realtime(uint64_t period, double busy,
+				     double busy_limit, bool preemptible)
+{
+	mach_timebase_info_data_t timebase_info;
+	mach_timebase_info(&timebase_info);
+
+	double realtime_period =
+		((double)timebase_info.denom / (double)timebase_info.numer) *
+		period;
+
+	thread_time_constraint_policy_data_t policy;
+	policy.period = (uint32_t)realtime_period;
+	policy.computation = (uint32_t)(realtime_period * busy);
+	policy.constraint = (uint32_t)(realtime_period * busy_limit);
+
+	policy.preemptible = preemptible;
+
+	int kr = thread_policy_set(pthread_mach_thread_np(pthread_self()),
+				   THREAD_TIME_CONSTRAINT_POLICY,
+				   (thread_policy_t)&policy,
+				   THREAD_TIME_CONSTRAINT_POLICY_COUNT);
+	if (kr != KERN_SUCCESS) {
+		blog(LOG_ERROR,
+		     "error setting rt thread, thread_policy_set: %s",
+		     mach_error_type(kr));
+	}
+
+	pthread_set_qos_class_self_np(QOS_CLASS_USER_INTERACTIVE, 0);
+}
+
 void *obs_graphics_thread_autorelease(void *param)
 {
+	// Set the priority of this thread for macOS for better timer precision
+	const double GRAPHICS_THREAD_BUSY = 0.5;
+	const double GRAPHICS_THREAD_BUSY_LIMIT = 1.0;
```

Comment:
I don't know if the OS is going to like period == constraint. That's like saying you may never give up the core. This isn't audio, but I wouldn't mind .75/.85 is here.

Reply:
And 15% might seem like a lot of slop, but OBS isn't a game where the render thread could need a lot of frame time. The two CPU-intensive tasks that OBS may do depending on configuration, FFmpeg swscale resizing and software encoding, happens on other threads. Assuming I'm not forgetting something.

Reply:
`swscale` should happen on the media-io or video thread iirc.

Reply:
@jpark37 From light testing, 0.75/0.85 appears to gain the same benefits under light load. Where the CPU is more contested the values will likely diverge but I agree that, in the short term, it's both safer from a regression standpoint and more polite the system as a whole to reduce the busy max guarantee

Reply:
Sounds good.